### PR TITLE
Add rule-transition diagrams in dot syntax

### PR DESCRIPTION
**Pull request by aniketd**
Motivation: Since,
1. there have been multiple occasions where we have liked to refer to the rule-transition diagram in the Shelley specification during our technical discussions, and
2. now for Conway such a diagram does not exist yet, and
3. we do not have replaced a prose-in-tex based format for specifications in Agda,

I thought it would be a good idea to keep these diagrams handy, "as graphviz dot code" (and its output) in version control for everyone's benefit.

**I may be completely wrong and we can close this PR if this is not very helpful.**

NOTE: It is still a draft, since I'm not entirely certain about the accuracy of the Conway rule transitions yet, and would like @WhatisRT or @Soupstraw 's review, especially on the rules that have been removed, etc.

NOTE: A good way to review the rendered SVG files is to click on the `raw` button to view it directly on the browser, instead of within the GitHub UI.

### CODE REVIEW

**Comment by aniketd** [comment]
Good additions (for each rule):
1. Environment
2. Signal
3. State

**Comment by aniketd** [comment]
TODO: Consider adding the 4 triggers in concensus that cause the rule transitions.

**Comment by aniketd** [comment]
Haven't been able to come to this one yet. The TODOs remain:
1. Convert all circles to boxes (trivial)
2. Add (Env, State, Signal) triple to all nodes in a readable manner (should be easy)
3. Add the _four triggers from consensus_ to consider the transition diagrams actually complete.

**Comment by aniketd** [comment]
Updating the transition diagram for Shelley with boxes and more details is left for later, when more bandwidth is available.

File: docs/graphviz-diagrams/conway_transitions.gv
Comment (Soupstraw) [inline docs/graphviz-diagrams/conway_transitions.gv]: We don't have those in Conway

**Review by Soupstraw — CHANGES_REQUESTED** [review]

File: docs/graphviz-diagrams/conway_transitions.gv
Comment (Soupstraw) [inline docs/graphviz-diagrams/conway_transitions.gv]:
```suggestion
```

File: docs/graphviz-diagrams/conway_transitions.gv
Comment (Soupstraw) [inline docs/graphviz-diagrams/conway_transitions.gv]:
```suggestion
```

**Review by lehins — APPROVED** [review]
LGTM.
Thank you for doing this

File: docs/graphviz-diagrams/conway_transitions.gv
Comment (lehins) [inline docs/graphviz-diagrams/conway_transitions.gv:1]: Don't merge until #3697 is merged

Comment (aniketd) [reply docs/graphviz-diagrams/conway_transitions.gv:1]: Noted! :memo:

**Comment by lehins** [comment]
@Soupstraw Need your unblock on this PR

**Review by Soupstraw — APPROVED** [review]

**Pull request merged**
